### PR TITLE
(PLATFORM-4513) Make InterwikiEdit work with both versions of MediaWiki

### PR DESCRIPTION
**Pull request by Grunny**
Make InterwikiEdit work with wikis on both versions of MediaWiki, so that the
current special page can be used to manage the links on all wikis while we are
migrating communities to the UCP.

This extension needs a rewrite, but since we will do so when porting it to the UCP,
I'm leaving it relatively untouched here.

/cc @Wikia/core-platform-team

### CODE REVIEW

**Review by mszabo-wikia — APPROVED** [review]
Lovely stuff

**Review by themech — APPROVED** [review]

File: extensions/wikia/WikiFactory/WikiFactory.php [hunk 1]
```diff
@@ -3616,4 +3616,16 @@ static public function unserializeHandler( $errno, $errstr ) {
...
 	}
+
+	/**
+	 * Temporary helper when we need to handle something for a UCP wiki
+	 * in the current app.
+	 *
+	 * @param  int     $wikiId
+	 * @return boolean
+	 */
+	public static function isUCPWiki( int $wikiId ): bool {
```
Comment (maaartyyynaa) [inline extensions/wikia/WikiFactory/WikiFactory.php]: I've created similiar function: https://github.com/Wikia/app/blob/dev/extensions/wikia/WikiFactory/WikiFactory.php#L3642-L3655, let's remove it or use it here too 

(If we stay with isSlot1Wiki, we could also fix comment `* Returns true if city_path for given wiki is equal as static::SLOT_1` -> `* Returns true if city_path for given wiki is equal to static::SLOT_1`)

**Review by maaartyyynaa — CHANGES_REQUESTED** [review]

File: extensions/wikia/SpecialInterwikiEdit/SpecialInterwikiEdit.php [hunk 2]
```diff
@@ -73,6 +73,15 @@ function execute( $par ){
 
 } // end class InterwikiEdit
 
+function wfSIWEMaybeAddExtraFields( int $wikiId, array $fields ): array {
```
Comment (harnash) [inline extensions/wikia/SpecialInterwikiEdit/SpecialInterwikiEdit.php]:
```suggestion
function wfSIWEAddExtraFields( int $wikiId, array $fields ): array {
```

File: extensions/wikia/SpecialInterwikiEdit/SpecialInterwikiEdit.php [hunk 3]
```diff
@@ -254,6 +263,7 @@ function wfSIWEEditInterwiki(){
 							</form>";
 					}else{
 						$ret .= "<p>Changeing interwiki...<br />\n";
+						$fields = wfSIWEMaybeAddExtraFields( $wikiaID, $fields );
```
Comment (harnash) [inline extensions/wikia/SpecialInterwikiEdit/SpecialInterwikiEdit.php]:
```suggestion
						$fields = wfSIWEAddExtraFields( $wikiaID, $fields );
```

File: extensions/wikia/SpecialInterwikiEdit/SpecialInterwikiEdit.php [hunk 4]
```diff
@@ -267,15 +277,19 @@ function wfSIWEEditInterwiki(){
...
-		        		$res = $db->query("INSERT INTO " .
-		        				"`interwiki` (iw_prefix,iw_url,iw_local,iw_trans) " .
-		        				"VALUES (". $db->addQuotes($fields['iw_prefix']). ",". $db->addQuotes($fields['iw_url']). ",".
-		        				$db->addQuotes($fields['iw_local']). ",". $db->addQuotes($fields['iw_trans']). ") ".
-		        				"ON DUPLICATE KEY ".
-		        				"UPDATE iw_url=". $db->addQuotes($fields['iw_url']).
-		        				", iw_local=". $db->addQuotes($fields['iw_local']).
-		        				", iw_trans=". $db->addQuotes($fields['iw_trans']), __METHOD__ );
+						$ret .= "<p>Changing interwiki...<br />\n";
+						$fields = wfSIWEMaybeAddExtraFields( $wikiaID, $fields );
```
Comment (harnash) [inline extensions/wikia/SpecialInterwikiEdit/SpecialInterwikiEdit.php]:
```suggestion
						$fields = wfSIWEAddExtraFields( $wikiaID, $fields );
```

File: extensions/wikia/SpecialInterwikiEdit/SpecialInterwikiEdit.php [hunk 5]
```diff
@@ -522,12 +536,22 @@ function wfSIWELinkWikisCommitProper ($linker, $linkee) {
...
-				" values('". $iw_prefix. "','". wfSIWEMakeInterlanguageUrl($linkee). "',1,0);");
-	} else {
-		return false;
+		$fields = [
+			'iw_prefix' => $iw_prefix,
+			'iw_url' => wfSIWEMakeInterlanguageUrl( $linkee ),
+			'iw_local' => '1',
+			'iw_trans' => '0',
+		];
+		$fields = wfSIWEMaybeAddExtraFields( $linker, $fields );
```
Comment (harnash) [inline extensions/wikia/SpecialInterwikiEdit/SpecialInterwikiEdit.php]:
```suggestion
		$fields = wfSIWEAddExtraFields( $linker, $fields );
```

**Review by harnash — CHANGES_REQUESTED** [review]

**Review by harnash — APPROVED** [review]

**Pull request merged**